### PR TITLE
Modular_exponentiation & binomial_coefficient_under_modulo in C++

### LINKED ISSUE #2299: [Cpp][Math/Number Theory] Modular Exponentiation
## 🚀 Feature

Modular exponentiation is used to compute pow(n1,n2)%mod (n1^n2%mod) in log(n2) time!

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes, I have read the contributing guidelines on PRs

## Motivation

I am a Competitive Programmer and I love Data Structures & Algorithms, I believe this algorithm is one of the basic yet one of the most used algorithms used. I want to contribute and add this algorithm to this repo, so in the future newcomers can refer to it and understand it.

## Pitch

As mentioned, this algorithm computes pow(n1,n2) in log(n2). The normal algorithm would take O(n2) times. This O(n2) is not feasible when n2 is larger than 10^7 or 10^8. So this algorithm is a must when dealing with large numbers!

### LINKED ISSUE #2328: [CPP][Math] Binomial Coefficient (under modulo)
## 🚀 Feature

Implement code to calculate NcR%mod in Cpp.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes, I have read the guidelines.

## Motivation & Pitch

Binomial coefficients are used in many problems and applications related to permutations and combinations (Example: Binomial coefficients are used to calculate Catalan Numbers)
There are some problems where numbers are very large, and we need to calculate these coefficients under some modulo to prevent overflow.
So, this CPP code will implement this feature only. Calculating binomial coefficient under modulo

### PR DESCRIPTION
**Pull request by Nkap23**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes, I have read the guidelines.

### Description

Added modular_exponentiation.cpp & binomial_coefficient_under_modulo.cpp in math section.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Resolves #2299 
Resolves #2328

### CODE REVIEW

File: C-Plus-Plus/math/modular_exponentiation.cpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+{
+    long long base,power,mod;
+    cout<<"Enter base: ";
+    cin>>base;
+    cout<<"Enter power: ";
+    cin>>power;
+    cout<<"Enter mod: ";
+    cin>>mod;
+    cout<<"power("<<base<<","<<power<<")%"<<mod<<" = "<<modular_exponentiation(base,power,mod)<<"\n";
+}
```
Comment (HarshCasper) [inline C-Plus-Plus/math/modular_exponentiation.cpp]: Add a new line at the EOF

File: C-Plus-Plus/math/modular_exponentiation.cpp [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+int main()
+{
+    long long base,power,mod;
+    cout<<"Enter base: ";
+    cin>>base;
+    cout<<"Enter power: ";
+    cin>>power;
+    cout<<"Enter mod: ";
+    cin>>mod;
+    cout<<"power("<<base<<","<<power<<")%"<<mod<<" = "<<modular_exponentiation(base,power,mod)<<"\n";
```
Comment (HarshCasper) [inline C-Plus-Plus/math/modular_exponentiation.cpp]: Use proper console statements

File: C-Plus-Plus/math/modular_exponentiation.cpp [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+{
+    base=base%mod;
+    long long answer=1;
+    while(power>0)
+    {
+        if(power%2==1) answer=(answer*base)%mod;
+        base=(base*base)%mod;
+        power=power/2;
+    }
+    return answer;
```
Comment (HarshCasper) [inline C-Plus-Plus/math/modular_exponentiation.cpp]: Format the Code

File: C-Plus-Plus/math/modular_exponentiation.cpp (hunk 1 above)
Comment (HarshCasper) [inline C-Plus-Plus/math/modular_exponentiation.cpp]: Add sample I/O

File: C-Plus-Plus/math/modular_exponentiation.cpp [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+We use this power of 2s to our advantage.
+(3^2)=(3^1)^2
+(3^4)=(3^2)^2 
+(3^8)=(3^4)^2 
+So, this reduces the number of multiplication, as we only need to sqaure the number (squaring is just single multiplication operation).
+
+So, overall
+Time Complexity: O(log(power))
+Space Complexity: O(1)
```
Comment (HarshCasper) [inline C-Plus-Plus/math/modular_exponentiation.cpp]: Shift complexity at the end of the code

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/math/modular_exponentiation.cpp (hunk 2 above)
Comment (Nkap23) [reply C-Plus-Plus/math/modular_exponentiation.cpp]: Can you please elaborate on what do you mean by "Use proper console statements".
Thank You!

Comment (HarshCasper) [reply C-Plus-Plus/math/modular_exponentiation.cpp]: Output Statements. Format them properly and make them as meaningful as possible

Comment (Nkap23) [reply C-Plus-Plus/math/modular_exponentiation.cpp]: Thank you for the clarification.
I have made the necessary changes.
Request you to review them.

**Review by HarshCasper — APPROVED** [review]

**Comment by Nkap23** [comment]
Sorry for the trouble. I wanted to create a new PR for my new issue (#2328), but when I pushed the changes to my branch, git automatically merged both my PRs.
What should I do?
I am now unable to separate these 2 PRs, should I delete this PR and create 2 new PR for both issues?
Thanks & Sorry
@HarshCasper

**Pull request merged**